dbcon/joblist/diskjoinstep.cpp

```diff
@@ -435,6 +437,12 @@ void DiskJoinStep::joinFcn()
                     outputDL->insert(joinResults[j]);
                 }
 
+                // MCOL-3879.  Clear the string store if it's accumulated > 50MB of mem
+                if (l_joinFERow.usesStringTable() && joinFEMem.getStringTableMemUsage() > 50 * (1 << 20))
+                    joinFEMem.clearStringStore();
+                if (baseRow.usesStringTable() && baseRowMem.getStringTableMemUsage() > 50 * (1 << 20))
+                    baseRowMem.clearStringStore();
```

Comment:
We should move this threshold value (50 * (1 << 20)) into a common location (maybe in rowgroup.h) as a const int or a macro, so it can be easily updated later.

Use of UNLIKELY around the if predicate might be a suitable optimization?

We are missing a line break before the if condition.